src/cdsetool/query.py

```diff
@@ -63,12 +63,73 @@ def __fetch_features(self):
         if self.next_url is not None:
             res = requests.get(self.next_url, timeout=120).json()
             self.features += res.get("features") or []
-
             total_results = res.get("properties", {}).get("totalResults")
             if total_results is not None:
                 self.total_results = total_results
-
+            odata =self.__fetch_odata(res.get("properties", {}))
+            if odata != {}:
+                self.__add_odata_to_features(odata)
```

Comment:
Here I have made a few changes, I like the idea of querying the odata in the query function. I have implemented two versions, if in the query_features function parameters we pass the True at the end we get the query with oData inside.
```Python
query_features(
    "Sentinel1",
    {
        "startDate": "2023-12-24",
        "completionDate": date(2024, 1, 4),
        "processingLevel": "LEVEL1",
        "sensorMode": "IW",
        "productType": "IW_GRDH_1S",
        # "orbitDirection":"dir",
        # "orbitNumber":13,
    },
    True #here the new parameter
)
```
Then since it might be useful for some people to check the odata I saved the oData inside a field in featues like this:
```Python
    def __add_odata_to_features(self,odata):
        """
        Given the odatadict, add the to correct (by matching the Ids) checksum the features
        """
        map_id_checksum = {}
        if "value" in odata:
            odata_list_value = odata["value"]
            for ordered_dict in odata_list_value:
                map_id_checksum[ordered_dict["Id"]] = {"Checksum": ordered_dict["Checksum"], "ContentLength": ordered_dict["ContentLength"],"odata": ordered_dict}
        for feature in self.features:
            if feature.get("id") in map_id_checksum.keys():
                ordered_dict = map_id_checksum[feature.get("id")]
                feature['odata'] = ordered_dict["odata"]
                feature['Checksum'] = ordered_dict["Checksum"]
                if feature.get('properties').get('services').get('download').get('size') == ordered_dict['ContentLength']:
                    feature['ContentLength'] = ordered_dict['ContentLength']
                else:
                    print(f"Warning: content length of {feature.get('id')} does not match the ContentLength in odata")
            else:
                print(f"Warning: {feature.get('id')} not in odata")
```
In which we still add the checksum and the oData (inside oData it will still be present the checksum). Let me know if you like this implementation or not. At the same time if you don't specify the "True" inside the query_featues I will fetch the oData during the checksum validate funtion like this:
```Python
def _checksum_compare(temp_path, product_info):
    """Compare a given MD5 checksum with one calculated from a file."""
    checksum = None
    algo = None
    filename = product_info.get("properties").get("title")
    if "Checksum" in product_info:
        # The intial query was made with the checksum option.
        checksum_list =product_info.get("Checksum")
        algo, checksum =from_checksum_list_to_checksum(checksum_list)
    else:
        odata_response = get_odata_from_product_title(filename)
        if 'value' in odata_response:
            if len(odata_response['value']) ==1:
                # odata['value'] is a list if there is more then one element for a single product
                # there is a error on the odata I guess 
                if 'Checksum' in odata_response['value'][0]:
                    checksum_list = odata_response['value'][0]['Checksum']
                    algo, checksum =from_checksum_list_to_checksum(checksum_list)
```
basically it checks wheter checksum is present in the options otherwise fetch the odata